### PR TITLE
add compressed fasta datatypes

### PR DESCRIPTION
**Pull request by matthdsm**
New datatype for compressed fasta references.
This ties is with your work for VEP, because VEP is impossibly slow when using an uncompressed reference.

Cheers
M

### CODE REVIEW

**Comment by matthdsm** [comment]
ready for merge!
Also added a fix for the `FastaDict` method
Cheers
M

**Comment by matthdsm** [comment]
Hi @illusional 
Do you suppose this one's good to go? Especially the RuntimeError fix could be useful, since it breaks the fastaDict type.
Cheers
M

File: janis_bioinformatics/data_types/fasta.py [hunk 1]
```diff
@@ -66,6 +67,69 @@ def secondary_files():
...
+    def __init__(self, optional=False):
+        super().__init__(optional, extension=".fa.gz")
+
+    @staticmethod
+    def name():
+        return "FastaGz"
+
+    def can_receive_from(self, other, source_has_default=False):
+
+        if isinstance(other, Fasta):
```
[the remark this answers is not in the document]
Comment (matthdsm) [reply janis_bioinformatics/data_types/fasta.py]: Thanks for the review! Should be a-ok now

**Comment by illusional** [comment]
Thanks @matthdsm, great addition!

**Comment by matthdsm** [comment]
My pleasure!

**Pull request merged**